src/react-app-env.d.ts

```diff
@@ -0,0 +1,15 @@
+/// <reference types="react-scripts" />
+import "styled-components";
+
+declare module "styled-components" {
```

Comment:
What is the general purpose of `.d.ts` files and what are the pros/cons? Is there any other way you could achieve the same result in typescript?

Reply:
I used the `.d.ts` file to modify types from the 'styled-components' library. It makes it easier to work with my custom theme in the future since it enables autocomplete for values listed in my custom type. This is the main pros. Regarding the cons, redefining types from external libraries is not the best idea.

>Is there any other way you could achieve the same result in typescript?

Good question. I haven't thought about it. Maybe, I could use `as const` for my theme object. Something like:
```
export const theme = {
  lightGrayColor: "#F5F5F5",
  darkGrayColor: "#424242",
  whiteColor: "#fff",
  orangeColor: "#EEB111",
  breakpoints: {
    tablet: (styles: FlattenSimpleInterpolation) =>
      `@media (min-width: 600px) {${styles}}`,
    desktop: (styles: FlattenSimpleInterpolation) =>
      `@media (min-width: 1025px) {${styles}}`,
  },
} as const;